process/estimator/executionResultInclusionEstimator.go

```diff
@@ -0,0 +1,191 @@
+package estimator
+
+import (
+	"math/bits"
+
+	"github.com/multiversx/mx-chain-go/config"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("process/executionResultInclusionEstimator")
+
+// Time per gas unit on minimum‑spec hardware (1 gas = 1ns)
+var tGas = uint64(1)
+
+// ExecutionResultMetaData is a lightweight summary EIE requires.
+type ExecutionResultMetaData struct {
+	HeaderHash   [32]byte // Link to full header in DB / cache
```

Comment:
maybe use []byte instead of [32], since we generally don’t define fixed byte array sizes in this project.